### ISSUE TITLE
ProjectType fails build when set to Class Library

### ISSUE THREAD

**Issue by situssoft** [Microsoft/VSProjectSystem#81]
I'm not sure where to raise issues for this SDK as its an odd hybrid of closed source code and open source docs. But here goes in the hope it has the desired effect,

If there is a better place to raise issues for the code rather than the Docs perhaps you could point me to the documentation of that place.

To reproduce the above:
Create a new "Project Type" project. 
Change nothing.
Press F5 to execute.
In the experimental instance create a new project of the new type.
Build the new project - all is well.
Change the project "Output Type" preperty to "Class Library"
Build the new project - error CSC : error CS2019: Invalid target type for /target: must specify 'exe', 'winexe', 'library', or 'module'

**Comment by adrianvmsft** [comment]
I took a look and it seems the problem is that the OutputType project property gets set to "dll" instead of "library" in the project which causes MSBuild to report that error.
I don't have yet a fix or workaround.
Thanks,
Adrian

**Comment by adrianvmsft** [comment]
It turns there are 2 issues causing this problem.

Until we release a new version of Project System Extensibility, here is how you can fix them in your project type:

1.  Project specific XAML rules can't override rules coming from MSBuild
To address this issue, you need to update `CustomProjectCs.targets` and switch the order of the 2 imports at the beginning such that CustomProject.targets follows Microsoft.CSharp.targets (which allows XAML rules defined in the project to override those that come from MSBuild).

  Before:
  ```xml
  <Import Project="$(MSBuildThisFileDirectory)\CustomProject.targets" />
  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
  ```

  After:
  ```xml
  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
  <!--Importing project specific targets after Microsoft.CSharp.targets so that project specific xaml rules can override those defined in MSBuild-->
  <Import Project="$(MSBuildThisFileDirectory)\CustomProject.targets" />
  ```

2. The value specified in `general.browseobject.xaml` is invalid. Replace `dll` with `library`

  Before:
  ```xml
  <EnumValue Name="dll" DisplayName="Class Library" />
  ```
  After:
  ```xml
  <EnumValue Name="library" DisplayName="Class Library" />
  ```

Thanks,
Adrian

**Comment by adrianvmsft** [comment]
Fixed in 14.1.127.50932 and newer builds. For existing project types, please reference to the workaround above.
